misk-hibernate/src/main/kotlin/misk/hibernate/HibernateModule.kt

```diff
@@ -72,8 +73,10 @@ class HibernateModule(
           sessionFactoryProvider.get(), config)
     }).asSingleton()
 
-    bind(transacterKey).toProvider(Provider<Transacter> {
-      RealTransacter(qualifier, sessionFactoryProvider.get(), config)
+    bind(transacterKey).toProvider(object : Provider<Transacter> {
+      @com.google.inject.Inject(optional=true) val tracer: Tracer? = null
```

Comment:
do we want to force misk apps to use guice?

Reply:
so does guice blow up if you try injecting a null value for a nullable type in kt?

Reply:
guice is already all over misk, my impression was that the two are coupled

Reply:
> you try injecting a null value for a nullable type in kt

Yea, tried locally and that's what I get

Reply:
well is guice impl all over or just javax.Inject? if we decide we want to be opinionated about guice, then we should just use the guice annotations everywhere IMO

Reply:
it does seem like it's opinionated. There's a class called Guice.